doc/configuration.rst

```diff
@@ -1116,6 +1117,105 @@ See the Sphinx-Gallery `Sphinx configuration file
 <https://github.com/sphinx-gallery/sphinx-gallery/blob/master/doc/conf.py>`_
 for an example that uses the `public Binder server <https://mybinder.org>`_.
 
+.. _jupyterlite:
+
+Generate JupyterLite links for gallery notebooks (experimental)
+===============================================================
+
+Sphinx-Gallery automatically generates Jupyter notebooks for any examples built
+with the gallery. `JupyterLite <https://jupyterlite.readthedocs.io>`__ makes it
+possible to run an example in your browser. The functionality is quite similar
+to Binder in the sense that you will get a Jupyter environment where you can
+run the example interactively as a notebook. The main difference with Binder
+are:
+
+- with JupyterLite, the example actually runs in your browser, there is no need
+  for a separate machine in the cloud to run your Python code. That means that
+  starting a Jupyter server is genrally quicker, no need to wait for the Binder
+  image to be built
+- with JupyterLite the first imports take time. At the time of writing
+  (February 2023) ``import scipy`` can take ~15-30s. Some innocuously looking
+  Python code may just not work and break in an unexpected fashion. The Jupyter
+  kernel is based on Pyodide, see some `here
+  <https://pyodide.org/en/latest/usage/wasm-constraints.html>`__ for some
+  Pyodide limitations.
+- with JupyterLite environments are not as flexible as Binder, for example you
+  can not use a docker image but only the default `Pyodide
+  <https://pyodide.org/en/stable/index.html>`__ environment. That means that
+  some non pure-Python packages may not be available, see list of `available
+  packages in Pyodide
+  <https://pyodide.org/en/stable/usage/packages-in-pyodide.html>`__.
```

Comment:
Maybe you could expand this point a bit and explain that this it's possible to install extra, pure-python packages from pypi.org using the following construct:

```python
try:
    # Manually install extra dependencies when running this example with
    # Pyodide:
    import micropip
    await micropip.install("package_name")
except ImportError:
    # Assume all the dependencies are already installed when running on other
    # types of hosts, such as binder.
    pass
```

Not protecting this snippet against `ImportError` would render the example pyodide-dependent as a result.

Reply:
Note that the `%pip` magic is also supported in the latest JupyterLite releases: https://jupyterlite.readthedocs.io/en/latest/howto/python/packages.html#installing-packages-at-runtime

```
%pip install package_name
```

Which could help make remove this `try`/ `except` block which could be awkward to some users.

Reply:
Nice I didn't know this!